### PR TITLE
Chore/update cennzx

### PR DESCRIPTION
**Pull request by jordy25519**

### CODE REVIEW

File: cli/src/cli.rs [hunk 1]
```diff
@@ -1,72 +1,38 @@
...
-// the Free Software Foundation, either version 3 of the License, or
-// (at your option) any later version.
-
-// Substrate is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-// GNU General Public License for more details.
-
-// You should have received a copy of the GNU General Public License
-// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (aliXsed) [inline cli/src/cli.rs:15]: Do we want to add our own header to this file?

File: cli/src/command.rs [hunk 2]
```diff
@@ -1,109 +1,156 @@
...
-				);
-				if let Err(e) = tracing::subscriber::set_global_default(subscriber) {
-					return Err(format!("Unable to set global default subscriber {}", e).into());
-				}
-			}
+	fn load_spec(&self, id: &str) -> Result<Box<dyn sc_service::ChainSpec>, String> {
+		Ok(match id {
+			// "dev" => Box::new(chain_spec::dev::config()?),
+			// "azalea" => Box::new(chain_spec::azalea::config()?),
+			// "nikau" => Box::new(chain_spec::nikau::config()?),
```
Comment (aliXsed) [inline cli/src/command.rs:60]: Can we enable these now?

Comment (jordy25519) [reply cli/src/command.rs:60]: Yup I've got a different PR to enable these: https://github.com/cennznet/cennznet/pull/289

File: crml/cennzx/src/lib.rs [hunk 3]
```diff
@@ -66,13 +64,44 @@ pub struct LiquidityPrice<Balance> {
...
 
-pub trait Trait: frame_system::Trait + pallet_generic_asset::Trait {
-	type Call: Parameter + Dispatchable<Origin = <Self as frame_system::Trait>::Origin>;
+pub trait Trait: frame_system::Trait {
+	/// The system event type
 	type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
-	/// A function type to get an exchange address given the asset ID pair.
-	type ExchangeAddressGenerator: ExchangeAddressFor<Self::AssetId, Self::AccountId>;
-	type BalanceToUnsignedInt: From<<Self as pallet_generic_asset::Trait>::Balance> + Into<LowPrecisionUnsigned>;
-	type UnsignedIntToBalance: From<LowPrecisionUnsigned> + Into<<Self as pallet_generic_asset::Trait>::Balance>;
```
Comment (aliXsed) [inline crml/cennzx/src/lib.rs:75]: Great that we don't need these anymore. Well done.

File: crml/cennzx/src/lib.rs [hunk 4]
```diff
@@ -66,13 +64,44 @@ pub struct LiquidityPrice<Balance> {
...
+	/// Type for identifying assets
+	type AssetId: Parameter + Member + AtLeast32BitUnsigned + Default + Copy + Into<u64>;
+	/// The type for asset amounts
+	/// Type for denoting asset balances
+	type Balance: Parameter + Member + AtLeast32BitUnsigned + Default + Copy;
+	/// Something which can provide asset management
+	type AssetSystem: SimpleAssetSystem<AccountId = Self::AccountId, Balance = Self::Balance, AssetId = Self::AssetId>;
```
Comment (aliXsed) [inline crml/cennzx/src/lib.rs]: Another nice improvement!

**Review by aliXsed — APPROVED** [review]
LVGTM!

**Comment by jordy25519** [comment]
Had this open to show the diff w/ develop, closing now for #287

**Pull request closed without merging**